nixos/modules/services/web-servers/uwsgi.nix

```diff
@@ -5,11 +5,24 @@ with lib;
 let
   cfg = config.services.uwsgi;
 
+  isEmperor = cfg.instance.type == "emperor";
+
+  imperialPowers =
+    [
+      # spawn other user processes
+      "CAP_SETUID" "CAP_SETGID"
+      "CAP_SYS_CHROOT"
+      # transfer capabilities
+      "CAP_SETPCAP"
+      # create other user sockets
+      "CAP_CHOWN"
```

Comment:
As someone who doesn't know much about `uwsgi` please forgive me if this is a stupid question, but won't granting `CAP_CHOWN` generally introduce more risk than running the `master` process as `root`? If you grant `CAP_CHOWN` to this process then the process will **always** be able to `chown` anything, even when running scripts which were intended to run without such privileges... won't it?

Reply:
I'm not how it could be worse than running as root: when root it will have the equivalent of CAP_SYS_ADMIN which includes rw access to all userspace memory and kernel mode code execution (via eBPF).
Note that these capabilities are only granted to the "emperor", which is basically a master process responsible of setting up "vassals". The vassals run as separate users after dropping privileges, including any capability, unless explicitely configured.

Reply:
Ah the privileges are only granted to a master process and not propagated to children processes? I thought `AmbientCapabilities` are propagated to children... 🤔

Reply:
They are, but uwsgi drops them automatically, unless you specify in the configuration which capabilities the child needs.

Reply:
Oh that's super convenient. Awesome! Great stuff @rnhmjoj 🎉